Source/DafnyCore/Backends/Dafny/DafnyCodeGenerator.cs

```diff
@@ -2035,75 +2179,204 @@ private class ExprLvalue : ILvalue {
           _ => null
         };
 
-        var opAst = op switch {
-          BinaryExpr.ResolvedOpcode.EqCommon => DAST.BinOp.create_Eq(
-            e0.Type.IsRefType,
-            !e0.Type.IsNonNullRefType
-          ),
-          BinaryExpr.ResolvedOpcode.NeqCommon => DAST.BinOp.create_Neq(
+        object B(_IBinOp binOp) {
+          return builder.Builder.BinOp((DAST.BinOp)binOp);
+        }
+
+        var opStringClosure = opString;
+        object C(System.Func<DAST.Expression, DAST.Expression, DAST.Expression> callback) {
+          return builder.Builder.BinOp(opStringClosure, callback);
+        }
+
+        var newBuilder = op switch {
+          BinaryExpr.ResolvedOpcode.EqCommon => B((BinOp)BinOp.create_Eq(
             e0.Type.IsRefType,
             !e0.Type.IsNonNullRefType
-          ),
-          BinaryExpr.ResolvedOpcode.Div => NeedsEuclideanDivision(resultType) ? DAST.BinOp.create_EuclidianDiv() : DAST.BinOp.create_Div(),
-          BinaryExpr.ResolvedOpcode.Mod => NeedsEuclideanDivision(resultType) ? DAST.BinOp.create_EuclidianMod() : DAST.BinOp.create_Mod(),
-          BinaryExpr.ResolvedOpcode.Imp => DAST.BinOp.create_Implies(),
-          BinaryExpr.ResolvedOpcode.InSet => DAST.BinOp.create_In(),
-          BinaryExpr.ResolvedOpcode.InSeq => DAST.BinOp.create_In(),
-          BinaryExpr.ResolvedOpcode.NotInSet => DAST.BinOp.create_NotIn(),
-          BinaryExpr.ResolvedOpcode.NotInSeq => DAST.BinOp.create_NotIn(),
-          BinaryExpr.ResolvedOpcode.SetDifference => DAST.BinOp.create_SetDifference(),
-          BinaryExpr.ResolvedOpcode.Concat => DAST.BinOp.create_Concat(),
-          _ => DAST.BinOp.create_Passthrough(Sequence<Rune>.UnicodeFromString(opString)),
+          )),
+          BinaryExpr.ResolvedOpcode.SetEq => B((BinOp)BinOp.create_Eq(false,false)),
```

Comment:
It would be so nice to generate all this boilerplate. Any chance you considered AutoExtern?

Reply:
Not considering AutoExtern anymore because the conversion is very specific now., as the intermediate language is no longer just Dafny but a subset of it For example, NeqCommon is translated to a Not Eq.